### PR TITLE
Implement defaults for end hosts

### LINKED ISSUE #37: Implement defaults for end hosts / servers
- fallback (port-channel)
- storm-control
- portfast

### PR DESCRIPTION
**Pull request by xaviramon**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Code style update (formatting, renaming)
- [x] Refactoring (no functional changes)
- [ ] Documentation content changes
- [ ] Other (please describe):

## Related Issue(s)

Fixes #37 
## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have updated [molecule CI](https://github.com/aristanetworks/ansible-avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly
- [ ] I have read the [**CONTRIBUTING**](https://www.avd.sh/docs/contributing/) document.
- [ ] All new and existing tests passed ([`pre-commit`](https://www.avd.sh/docs/installation/development/#python-virtual-environment), `make linting` and `make sanity-lint`).

### CODE REVIEW

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/ethernet-interfaces.j2 [hunk 1]
```diff
@@ -46,10 +46,10 @@ interface {{ ethernet_interface }}
 {%             if ethernet_interfaces[ethernet_interface].qos.trust is defined and ethernet_interfaces[ethernet_interface].qos.trust is not none %}
    qos trust {{ ethernet_interfaces[ethernet_interface].qos.trust }}
 {%             endif %}
-{%             if ethernet_interfaces[ethernet_interface].spanning_tree_portfast is defined and ethernet_interfaces[ethernet_interface].spanning_tree_portfast == 'edge' %}
+{%             if ethernet_interfaces[ethernet_interface].spanning_tree_portfast is defined and ethernet_interfaces[ethernet_interface].spanning_tree_portfast is none %}
    spanning-tree portfast
-{%             elif ethernet_interfaces[ethernet_interface].spanning_tree_portfast is defined and ethernet_interfaces[ethernet_interface].spanning_tree_portfast == 'network' %}
-   spanning-tree portfast network
+{%             elif ethernet_interfaces[ethernet_interface].spanning_tree_portfast is defined and ethernet_interfaces[ethernet_interface].spanning_tree_portfast is not none %}
```
Comment (titom73) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/ethernet-interfaces.j2]: Maybe we can check user configured correct value with a test like:

```jinja2
{% if ethernet_interfaces[ethernet_interface].spanning_tree_portfast in ['auto', 'edge', ''network, 'normal'] %}
```

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/port-channel-interfaces.j2 [hunk 2]
```diff
@@ -50,10 +50,10 @@ interface {{ port_channel_interface }}
 {%         if port_channel_interfaces[port_channel_interface].qos.trust is defined and port_channel_interfaces[port_channel_interface].qos.trust is not none %}
    qos trust {{ port_channel_interfaces[port_channel_interface].qos.trust }}
 {%         endif %}
-{%         if port_channel_interfaces[port_channel_interface].spanning_tree_portfast is defined and port_channel_interfaces[port_channel_interface].spanning_tree_portfast == 'edge' %}
+{%         if port_channel_interfaces[port_channel_interface].spanning_tree_portfast is defined and port_channel_interfaces[port_channel_interface].spanning_tree_portfast is none %}
    spanning-tree portfast
-{%         elif port_channel_interfaces[port_channel_interface].spanning_tree_portfast is defined and port_channel_interfaces[port_channel_interface].spanning_tree_portfast == 'network' %}
-   spanning-tree portfast network
+{%         elif port_channel_interfaces[port_channel_interface].spanning_tree_portfast is defined and port_channel_interfaces[port_channel_interface].spanning_tree_portfast is not none %}
```
Comment (titom73) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/port-channel-interfaces.j2]: Maybe we can check user configured correct value with a test like:

```jinja2
{% if ethernet_interfaces[ethernet_interface].spanning_tree_portfast in ['auto', 'edge', ''network, 'normal'] %}
```

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/port-channel-interfaces.j2 [hunk 3]
```diff
@@ -105,6 +113,13 @@ interface {{ port_channel_interface }}
    pim ipv4 sparse-mode
 {%            endif %}
 {%         endif %}
+{%         if port_channel_interfaces[port_channel_interface].storm_control is defined and port_channel_interfaces[port_channel_interface].storm_control is not none %}
```
Comment (titom73) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/port-channel-interfaces.j2:125]: Change already covered by PR https://github.com/aristanetworks/ansible-avd/pull/306

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/ethernet-interfaces.j2 [hunk 4]
```diff
@@ -132,6 +132,13 @@ interface {{ ethernet_interface }}
 {%             if ethernet_interfaces[ethernet_interface].pim.ipv4.sparse_mode is defined and ethernet_interfaces[ethernet_interface].pim.ipv4.sparse_mode == true %}
    pim ipv4 sparse-mode
 {%             endif %}
+{%             if ethernet_interfaces[ethernet_interface].storm_control is defined and ethernet_interfaces[ethernet_interface].storm_control is not none %}
```
Comment (titom73) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/ethernet-interfaces.j2:141]: Change already covered by PR https://github.com/aristanetworks/ansible-avd/pull/306

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/port-channel-interfaces.j2 [hunk 5]
```diff
@@ -88,6 +88,14 @@ interface {{ port_channel_interface }}
 
 {%                 endfor %}
 {%             endif %}
+{%         if port_channel_interfaces[port_channel_interface].lacp_fallback is defined and port_channel_interfaces[port_channel_interface].lacp_fallback is not none %}
```
Comment (titom73) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/port-channel-interfaces.j2:100]: Already covered by https://github.com/aristanetworks/ansible-avd/pull/368

**Review by titom73 — CHANGES_REQUESTED** [review]
Please update code to remove LACP fallback (https://github.com/aristanetworks/ansible-avd/pull/368) and Storm-control (#306)

**Comment by titom73** [comment]
PR to be updated with:

- [ ] Edge STP modes
- [ ] Storm control for port-channel (to be tested)

**Comment by carlbuchmann** [comment]
closing due to stale PR

**Pull request closed without merging**